addons/account/wizard/account_payment_register.py

```diff
@@ -124,14 +124,17 @@ def _get_line_batch_key(self, line):
         will be grouped together.
         :return: A python dictionary.
         '''
-        return {
+        res = {
             'partner_id': line.partner_id.id,
             'account_id': line.account_id.id,
             'currency_id': (line.currency_id or line.company_currency_id).id,
             'partner_bank_id': line.move_id.partner_bank_id.id,
             'partner_type': 'customer' if line.account_internal_type == 'receivable' else 'supplier',
             'payment_type': 'inbound' if line.balance > 0.0 else 'outbound',
         }
+        if line.move_id.partner_bank_id.id:
```

Comment:
you mean is_inbound() ?

Reply:
no: outbond

```
200 partner_bank_id = fields.Many2one('res.partner.bank', string='Recipient Bank',
201 help='Bank Account Number to which the invoice will be pai**d. A Company bank account if this is a Customer Invoice or Vendor Credit Note**, otherwise a Partner bank account number.',
```
? am I missing something?

Reply:
The goal is, if a partner_bank_id is set on **an invoice**, and this bank account is also set on a bank journal, to set said journal as journal_id of the payment.

Reply:
@qdp-odoo (ping, to get trough your gmail filtering), see above

Reply:
> The goal is, if a partner_bank_id is set on **an invoice**, and this bank account is also set on a bank journal, to set said journal as journal_id of the payment. (Instead of default first bank or cash journal)

my goal is to avoid a useless search in case of a **supplier invoice**. Given that the bank account that will be set on such an invoice will belong to the vendor, not the company, there is no point entering the `if`

FYI: the first message got through the mail filtering ;-)

Reply:
> > The goal is, if a partner_bank_id is set on **an invoice**, and this bank account is also set on a bank journal, to set said journal as journal_id of the payment. (Instead of default first bank or cash journal)
> 
> my goal is to avoid a useless search in case of a **supplier invoice**. Given that the bank account that will be set on such an invoice will belong to the vendor, not the company, there is no point entering the `if`

I have the feeling that we are not talking about the same thing.
If a move is a Customer Invoice or a Vendor Refund, the partner_bank_id will be a company bank account. That's when we want to search. Those are inbound.
If a move is a Customer Credit Note or a Vendor Bill, the partner_bank_id will be a vendor account. That's when we do not want to search. Those are outbound.
So we want to search the journal if the move is inbound.
> FYI: the first message got through the mail filtering ;-)

Sorry 😬

Reply:
we are talking about the same thing, I simply trusted my guts (without checking) on the wording of those methods is_outbound() and is_inbound(). I do not get why inbound would be related to out_invoice and in_refund, that feels counter-intuitive.

Well, anyway, you were right since the beginning :D Sorry about that!